androidApp/src/main/java/com/mbta/tid/mbta_app/android/stopDetails/StopDetailsViewModel.kt

```diff
@@ -534,15 +537,23 @@ fun stopDetailsManagedVM(
     val groupByDirection by viewModel.groupByDirection.collectAsState()
     val departures by viewModel.stopDepartures.collectAsState()
 
+    var wasInBackground by remember { mutableStateOf(false) }
+
     LaunchedEffect(stopId) { viewModel.handleStopChange(stopId) }
     LaunchedEffect(filters?.tripFilter) { viewModel.handleTripFilterChange(filters?.tripFilter) }
 
     LifecycleResumeEffect(null) {
-        viewModel.returnFromBackground()
-        viewModel.rejoinStopPredictions()
-        viewModel.rejoinTripChannels()
+        // we don’t want to return and rejoin on first appearance, only on actual resume, because
+        // the handleStopChange/handleTripFilterChange will handle initial joins
```

Comment:
👍 thought (non-blocking): I can't remember why I initially created separate `rejoinStopPredictions` and `rejoinTripChannels` functions which don't have an equivalent in the iOS implementation. It seems possible that condensing these three effects into a single `LifecycleResumeEffect` that takes stopId and `filters?.tripFilter` could help to simplify things here and make it less error prone.

Reply:
It’d be a little weird to be calling `handleStopChange` if the stop filter changes but the trip filter doesn‘t, but I think having separate `LifecycleResumeEffect`s for the stop and the trip Should™ work.